### PR TITLE
Python: CJK support for text splitter

### PR DESCRIPTION
**Pull request by marlenezw**
### Motivation and Context
The text splitting algorithm has hardcoded values for punctuation based on ASCII punctuation (,.; etc.) Chinese and Japanese use both full-width Unicode punctuation values https://en.wikipedia.org/wiki/Halfwidth_and_Fullwidth_Forms_(Unicode_block) or ideographic punctuation.
If you give the text splitter Japanese text it will incorrectly split in the middle of sentences
The current algorithm defaults to an approximation token counter, which is partly accurate for Indo-European/Latin Alphabet languages but totally out for CJK. This tests uses the BPE token length for cl100k_base (GPT3/4)

This is the Python version of [this PR](https://github.com/microsoft/semantic-kernel/pull/5489)

### Description

This change adds some basic full-width characters and ideographic punctuation to the built-in list.
It also adds a test which checks that it splits a sentence correctly. The first part of the test string (田中の猫はかわいいですね.) is 16 tokens for cl100k_base.

### Contribution Checklist

 The code builds clean without any errors or warnings
The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
All unit tests pass, and I have added new tests where possible
I didn't break anything!

### CODE REVIEW

**Comment by markwallace-microsoft** [comment]
<a href="https://github.com/microsoft/semantic-kernel/blob/744c7635464d0466bf4f7d7593c6def37aaa0ead/README.md">Py3.8 Test Coverage</a><details><summary>Python 3.8 Test Coverage Report • </summary><table><tr><th>File</th><th>Stmts</th><th>Miss</th><th>Cover</th><th>Missing</th></tr><tbody><tr><td colspan="5"><b>semantic_kernel/text</b></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/semantic-kernel/blob/744c7635464d0466bf4f7d7593c6def37aaa0ead/python/semantic_kernel/text/text_chunker.py">text_chunker.py</a></td><td>113</td><td>4</td><td>96%</td><td><a href="https://github.com/microsoft/semantic-kernel/blob/744c7635464d0466bf4f7d7593c6def37aaa0ead/python/semantic_kernel/text/text_chunker.py#L208">208</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/744c7635464d0466bf4f7d7593c6def37aaa0ead/python/semantic_kernel/text/text_chunker.py#L231">231</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/744c7635464d0466bf4f7d7593c6def37aaa0ead/python/semantic_kernel/text/text_chunker.py#L248">248</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/744c7635464d0466bf4f7d7593c6def37aaa0ead/python/semantic_kernel/text/text_chunker.py#L308">308</a></td></tr><tr><td><b>TOTAL</b></td><td><b>5365</b></td><td><b>987</b></td><td><b>82%</b></td><td>&nbsp;</td></tr></tbody></table></details>

## Python 3.8 Unit Test Overview
| Tests | Skipped | Failures | Errors | Time |
| ----- | ------- | -------- | -------- | ------------------ |
| 1208 | 11 :zzz: | 0 :x: | 0 :fire: | 15.992s :stopwatch: |

**Comment by marlenezw** [comment]
Adding [related issue] (https://github.com/microsoft/semantic-kernel/issues/5490)

File: python/pyproject.toml [hunk 1]
```diff
@@ -87,6 +87,7 @@ azure-core = "^1.28.0"
 transformers = "^4.28.1"
 sentence-transformers = "^2.2.2"
 torch = "^2.2.0"
+tiktoken = "^0.6.0"
```
Comment (marlenezw) [inline python/pyproject.toml:90]: @tonybaloney  I added tiktoken to the unit test dependencies. I'll open a separate issue for Jinja after reading the PR where it was added.

File: python/semantic_kernel/text/text_chunker.py [hunk 2]
```diff
@@ -13,23 +13,23 @@
 
 TEXT_SPLIT_OPTIONS = [
     ["\n", "\r"],
-    ["."],
+    [".", "。", "．"],
```
Comment (marlenezw) [inline python/semantic_kernel/text/text_chunker.py:16]: Updated as suggested on the previous PR here and everywhere else.

**Review by tonybaloney — APPROVED** [review]

**Comment by moonbox3** [comment]
Hi @marlenezw, thanks for working on this. Could you please pull latest main, and re-build the poetry lock file from the Python directory? (`poetry lock --no-update`)

**Comment by moonbox3** [comment]
Hi @marlenezw, since we haven't heard from you in several weeks, we will close the PR. When ready, please re-open the PR against latest code from main. Thank you.

**Pull request closed without merging**